node/modules/lp2p/rcmgr.go

```diff
@@ -0,0 +1,66 @@
+package lp2p
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"go.uber.org/fx"
+
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p-core/network"
+	rcmgr "github.com/libp2p/go-libp2p-resource-manager"
+
+	"github.com/filecoin-project/lotus/node/repo"
+)
+
+func ResourceManager(lc fx.Lifecycle, repo repo.LockedRepo) (network.ResourceManager, error) {
+	var limiter *rcmgr.BasicLimiter
+	var opts []rcmgr.Option
+
+	repoPath := repo.Path()
+
+	// create limiter -- parse $repo/limits.json if exists
+	limitsFile := filepath.Join(repoPath, "limits.json")
```

Comment:
Feels like something that should be a part of the libp2p config section, not it's own file.

https://github.com/filecoin-project/lotus/blob/c0f29ea9c6d188f4b4dc997a287bcdb560af7f48/node/config/types.go#L303

Reply:
sure, can do.

Reply:
oh you want to embed the limits in the config?
I am not sure if it is such a great idea, but i could export the struct used by the parser to embed it.

However, i really prefer to have it separate for easy sharing and modification.